ipa/scripts/cli.py

```diff
@@ -174,6 +174,14 @@ def main(context, no_color):
     '--results-dir',
     help='Directory to store test results and output.'
 )
+@click.option(
+    '--root-device-size',
+    help='The size in GB of the root disk device.'
+)
+@click.option(
+    '--root-device-type',
+    help='The device type of the root disk.'
+)
```

Comment:
Any will do, I don't think we should expose this on the command line. We should just let these come from a test.

Reply:
Does it hurt to have it exposed? If you set the size for any provider you also are required to set a type which means we would need a default on creation. Thus I think it's nice to allow both to be overridden.

Reply:
> The test should run an instance where the root disk is set to larger than the default. (From email)

If the instance is launched with a disk larger than default then it must be provided by a command line arg/flag. Because the instance is launched first before checking any tests.

A re-size test would separate of this.

Reply:
Well I don't really want to clutter up the command line with stuff that is really specialized I'd like to avoid having the IPA command line look like the qemu command line, which can easily happen if we take all possible options into account.

I guess I was more thinking along the lines of a "re-size" test, sorry if I didn't express that properly.

Reply:
Okay, yes I was thinking we wanted both (at launch and resize). Even if this is only related to growing the disk I think the simplest solution is to use a command line option.

If this is a test then I don't think it even fits as part of the IPA test suite. Since this would require credential handling and a separate test for each provider I think it's implementation specific. It would require a lot of re-tooling IPA to pass credentials to individual test modules.

Reply:
Well we do not really need credentials for individual tests, we need API access that is provided by Roles in EC2, a switch in GCE, and I don't know how it works in IPA.

But you are correct the idea of fiddling with the instance and using the framework API does not really with with the concept of IPA. WE should probably have an IPA external test suite for that.

OK if we just drop this?

Reply:
Yes this can be dropped. We will need to discuss the options available to test the expanded root disk.